index.html

```diff
@@ -935,9 +934,9 @@ <h2>
           then return <a>a promise rejected with</a> an
           "<a>InvalidStateError</a>" <a>DOMException</a>.
           </li>
-          <li>If the <a>user agent</a>'s <a>payment request is showing</a>
-          boolean is true, then return <a>a promise rejected with</a> an
-          "<a>AbortError</a>" <a>DOMException</a>.
+          <li>If the <a>top-level browsing context</a>'s <a>payment request is
```

Comment:
"the top-level browsing context" is not well-defined. Consider a same-origin popup window (i.e. two separate top-level windows which can synchronously access each other) in which you call `window1.PaymentRequest.prototype.show.call(window2PaymentRequestInstance)`.

You want "this `PaymentRequest`'s [relevant global object](https://html.spec.whatwg.org/#concept-relevant-global)'s browsing context's top-level browsing context". Maybe make a definition to shorten that (e.g. "payment-relevant browsing context of _obj_" or "payment request is showing in the same browsing context as _obj_").

And, if you're feeling evil, write tests for the case I described ^_^.

Reply:
Thanks! I had a bad feeling about "top-level browsing context"... the suggestion is great. I'll definitely add the test 😼.

Reply:
> `window1.PaymentRequest.prototype.show.call(window2PaymentRequestInstance)` 

@domenic, just to make sure I got this right. 

Both of these would theoretically show the payment sheet in the popup: 

```JS
button.onclick = () => {
    const popup = window.open("some.html");
    await new Promise(resolve => (popup.onload = resolve));
    const r = new popup.PaymentRequest(methods, details);
    await r.show();
}
```
And: 

```JS
button.onclick = () => {
    const popup = window.open("some.html");
    await new Promise(resolve => (popup.onload = resolve));
    const r = new popup.PaymentRequest(methods, details);
    await PaymentRequest.prototype.show.call(r);
}
```

Is that correct? .... 

And then could go from `popup.opener`, and show a payment sheet in the opener from the popup.  

I've put up an initial set that use just an iframe:
https://github.com/w3c/payment-request/pull/811

Reply:
Err, sorry, copypasta, tests are here  https://github.com/web-platform-tests/wpt/pull/14297
(I've not run them - so probably buggy)

Reply:
> Both of these would theoretically show the payment sheet in the popup:

Yeah, that's right!

> And then could go from popup.opener, and show a payment sheet in the opener from the popup.

I don't quite understand what you mean here.

> I've put up an initial set that use just an iframe:

I'll take a look, but iframes are not as good a test here, as they aren't top-level browsing contexts. The payment-relevant browsing context of a PaymentRequest created inside an iframe is its parent browsing context. Popups are special because they are top-level; they don't have a parent, just an opener.

Reply:
> And then could go from popup.opener, and show a payment sheet in the opener from the popup.

@domenic, imagine the following situation:

```JS
// this is actually a new browser tab...
const popupTab = window.open("tab.html");

// in which tab does the sheet show? In the popup tab or in opener tab? 
await new popupTab.PaymentRequest(method, details).show();
```

I guess "implementation detail"... 

Similarly, consider this one (little window):

```JS
// We make a little window... imagine, without toolbars, etc.
const popupWindow = window.open("window.html", "width=100,height=100,...");

// in which window does the sheet show? In the little window or in opener? 
await new popupWindow.PaymentRequest(method, details).show();
```

Similarly, I guess "implementation detail"?... don't think we can do much in the spec, but just wanted to point this out.